### PR TITLE
New post: large numbers in SKI and Xi function records

### PR DESCRIPTION
**Pull request by komiamiko**
Using new methods, I named large numbers in SKI and set greatly improved lower bounds on the Xi0 function.

By nature of the post, there isn't much to review in terms of content, though please do comment on anything notable or which needs changing.

I can't expect my reviewers to all understand ordinals very well, but please at least check that the combinator counts are correct. I did my best to ensure all information is accurate, and I will be doing some recounts myself, but still I am just a little worried I've made a mistake around the w^w level due to using `ski_tools.py` before the parser bug was fixed.

Also, please play around with `ski_tools.py` and ensure there's no obvious bugs.

### CODE REVIEW

**Comment by komiamiko** [comment]
Actually I should be more worried that I slipped up in the process of translating some of the larger functions to SKI, especially for the Goodstein sequence ones, where a mistake is likely to go unnoticed. The script does its best, and I don't think the bug actually affected me since I worked only in normal form (minimal brackets) as I recall.

File: _posts/2020-06-21-ski-numerals.markdown [hunk 1]
```diff
@@ -0,0 +1,923 @@
...
+As a reminder, this is what each combinator does:
+
+```
+Sxyz = xz(yz)
+Kxy = x
+Ix = x
+```
+
+What we informally call "evaluation" of these rules is formally called "beta-reduction".
+One would beta-reduce an SKI expression until no further rules can be applied,
```
Comment (simcard0000) [inline _posts/2020-06-21-ski-numerals.markdown:75]: "beta-reduce **a** SKI expression"

File: _posts/2020-06-21-ski-numerals.markdown [hunk 2]
```diff
@@ -0,0 +1,923 @@
...
+which occurs when none of the combinators have enough arguments to use their rule.
+Conventionally, we beta-reduce at the leftmost possible sub-expression.
+For example, starting with \\\(SSS(SI)S\\\), here is a the order of steps until it is maximally beta-reduced:
```
Comment (simcard0000) [inline _posts/2020-06-21-ski-numerals.markdown:78]: "here is the order of steps" (taking out the extra "a")

File: _posts/2020-06-21-ski-numerals.markdown [hunk 3]
```diff
@@ -0,0 +1,923 @@
...
+{: #building-goodstein}
+
+{% include collapser.markdown %}
+
+<div>
+
+At a time like this, I really wish I had an optimizing SKI compiler.
+Here I am doing my best to hand optimize it, ignoring numerous other branches in the tree of possibilities which could possibly lead to a more optimized version, just taking the smallest expression I can get for each part and then linking with its dependencies.
+
+I am by no means claiming the final expression I get is optimal
```
Comment (simcard0000) [inline _posts/2020-06-21-ski-numerals.markdown:822]: A period is needed at the end of the sentence: "...I get is optimal."

File: _posts/2020-06-21-ski-numerals.markdown (hunk 1 above)
Comment (komiamiko) [reply _posts/2020-06-21-ski-numerals.markdown:75]: isn't it ess-kay-eye when you say it though, so effectively starting with a vowel

File: _posts/2020-06-21-ski-numerals.markdown (hunk 2 above)
Comment (komiamiko) [reply _posts/2020-06-21-ski-numerals.markdown:78]: good catch!

File: _posts/2020-06-21-ski-numerals.markdown (hunk 3 above)
Comment (komiamiko) [reply _posts/2020-06-21-ski-numerals.markdown:822]: nice work spotting that, thank you

**Pull request merged**